src/coreclr/vm/comutilnative.cpp

```diff
@@ -1624,7 +1624,8 @@ BOOL CanCompareBitsOrUseFastGetHashCode(MethodTable* mt)
     }
 
     if (mt->ContainsPointers()
-        || mt->IsNotTightlyPacked())
+        || mt->IsNotTightlyPacked()
+        || mt->GetClass()->IsInlineArray())
```

Comment:
I _think_ this might need a matching change on native AOT side (either in RequiresValueTypeGetFieldHelperMethod or `ComparerIntrinsics.CanCompareValueTypeBits`), otherwise we'll have different behavior for InlineArray of size 1 (with e.g. Int32 field) since those typically don't have gaps in the layout and are eligible to be compared by hashing/comparing bits.

Or we could say that InlineArray of size 1 that doesn't contain pointers and is tightly packed has a working Equals/GetHashCode and drop the addition here.

Reply:
Indeed. Looks like I'm missing a test case here.

Reply:
@MichalStrehovsky Thanks for noting this. Added test and sure enough a check was missing.